bcipy/signal/model/offline_analysis.py

```diff
@@ -224,9 +214,263 @@ def offline_analysis(
+
+
+@report_execution_time
+def offline_analysis(
+    data_folder: str = None,
+    parameters: Parameters = None,
+    alert_finished: bool = True,
+    estimate_balanced_acc: bool = False,
+    show_figures: bool = False,
+    save_figures: bool = False,
+) -> Tuple[SignalModel, Figure]:
+    """Gets calibration data and trains the model in an offline fashion.
+    pickle dumps the model into a .pkl folder
+
+    How it Works:
+    - reads data and information from a .csv calibration file
+    - reads trigger information from a .txt trigger file
+    - filters data
+    - reshapes and labels the data for the training procedure
+    - fits the model to the data
+        - uses cross validation to select parameters
+        - based on the parameters, trains system using all the data
+    - pickle dumps model into .pkl file
+    - generates and [optional] saves/shows the ERP figure
+    - [optional] alert the user finished processing
+
+    Parameters:
+    ----------
+        data_folder(str): folder of the data
+            save all information and load all from this folder
+        parameter(dict): parameters for running offline analysis
+        alert_finished(bool): whether or not to alert the user offline analysis complete
+        estimate_balanced_acc(bool): if true, uses another model copy on an 80/20 split to
+            estimate balanced accuracy
+        show_figures(bool): if true, shows ERP figures after training
+        save_figures(bool): if true, saves ERP figures after training to the data folder
+
+    Returns:
+    --------
+        model (SignalModel): trained model
+        figure_handles (Figure): handles to the ERP figures
+    """
+    assert parameters, "Parameters are required for offline analysis."
+    if not data_folder:
+        data_folder = load_experimental_data()
+
+    devices_by_name = devices.load(
+        Path(data_folder, DEFAULT_DEVICE_SPEC_FILENAME), replace=True)
+    data_file_paths = [
+        path for path in (Path(data_folder, raw_data_filename(device_spec))
+                          for device_spec in devices_by_name.values())
+        if path.exists()
+    ]
+
+    for raw_data_path in data_file_paths:
+        raw_data = load_raw_data(raw_data_path)
+        device_spec = devices_by_name.get(raw_data.daq_type)
+        # extract relevant information from raw data object eeg
+        if device_spec.content_type == "EEG":
+            analyze_erp(raw_data, parameters, device_spec, data_folder, estimate_balanced_acc,
+                        save_figures, show_figures)
+
+        if device_spec.content_type == "Eyetracker":
+            analyze_gaze(raw_data, device_spec, data_folder, save_figures, show_figures,
+                         model_type="Centralized")
+
     if alert_finished:
         play_sound(f"{STATIC_AUDIO_PATH}/{parameters['alert_sound_file']}")
-    return model, figure_handles
+    # return model, figure_handles
+    return
```

Comment:
This should return a model(s) and any figure handles. Do the integration tests still pass with this change?